daemon/daemon.go

```diff
@@ -724,26 +610,17 @@ func NewDaemon(config *Config, registryService *registry.Service) (daemon *Daemo
 	if config.Mtu == 0 {
 		config.Mtu = getDefaultNetworkMtu()
 	}
-	// Check for mutually incompatible config options
-	if config.Bridge.Iface != "" && config.Bridge.IP != "" {
-		return nil, fmt.Errorf("You specified -b & --bip, mutually exclusive options. Please specify only one.")
-	}
-	if !config.Bridge.EnableIptables && !config.Bridge.InterContainerCommunication {
-		return nil, fmt.Errorf("You specified --iptables=false with --icc=false. ICC uses iptables to function. Please set --icc or --iptables to true.")
-	}
-	if !config.Bridge.EnableIptables && config.Bridge.EnableIpMasq {
-		config.Bridge.EnableIpMasq = false
+
+	// Ensure we have compatible configuration options
+	if err := checkConfigOptions(config); err != nil {
```

Comment:
@jhowardmsft why is this platform specific? seems like the code removed on the left is exactly duplicated twice.